src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -10490,6 +10490,115 @@ public void SynthesizedDelegateTypes_NamedArguments_MoreThan16Parameters()
                 Diagnostic(ErrorCode.ERR_NoCorrespondingArgument, "inferred").WithArguments("arg1", "<anonymous delegate>").WithLocation(7, 1));
         }
 
+        [Fact]
+        public void SynthesizedDelegateTypes_MoreThan16Parameters_DefaultParameterValue()
+        {
+            var range = Enumerable.Range(1, 17);
+            var manyParams = string.Join(", ", range.Select(i => $"int p{i}"));
+            var manyTypes = string.Join(",", range.Select(_ => "System.Int32"));
+            var source = $$"""
+                using System;
+                static void Report(Delegate d) => Console.WriteLine(d.GetType());
+                var lam1 = ({{manyParams}} = 1) => { };
+                Report(lam1);
+                var lam2 = ({{manyParams}} = 1) => { };
+                Report(lam2);
+                var lam3 = ({{manyParams}} = 2) => { };
+                Report(lam3);
+                """;
+            CompileAndVerify(source, expectedOutput: $"""
+                <>f__AnonymousDelegate0`17[{manyTypes}]
+                <>f__AnonymousDelegate0`17[{manyTypes}]
+                <>f__AnonymousDelegate1`17[{manyTypes}]
+                """).VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void SynthesizedDelegateTypes_MoreThan16Parameters_ParamsArray()
+        {
+            var range = Enumerable.Range(1, 16);
+            var manyParams = string.Join(", ", range.Select(i => $"int p{i}"));
+            var manyTypes = string.Join(",", range.Select(_ => "System.Int32"));
+            var source = $$"""
+                using System;
+                static void Report(Delegate d) => Console.WriteLine(d.GetType());
+                var lam1 = ({{manyParams}}, params int[] xs) => { };
+                Report(lam1);
+                var lam2 = ({{manyParams}}, params int[] xs) => { };
```

Comment:
>int [](http://example.com/codeflow?start=51&length=3)

Let's test if we are going to unify across different element types.

Reply:
Great idea, thanks, added.